### PR TITLE
Re-organize clustering module README and add section on colData contents

### LINKED ISSUE #294: Reorganize clustering module docs to mirror GOI module docs
**Please provide some background on the proposed additions or changes.**

We will want to modify the clustering module README.md structure to mirror that of the genes of interest module docs.

**What are the changes that you are proposing?**

We will want to re-organize the current clustering module README.md file so that the following is true:
- The table of contents should be first, followed by the analysis overview section
- The "running the workflow" section should be next
- Then expected input, parameters, and expected output

**Please describe the proposed solution.**

Using the GOI module README as a guide, we should restructure the clustering module README to be parallel to that.

### PR DESCRIPTION
**Pull request by cbethell**
**Issue Addressed**
Closes #294 and #240

**What is the purpose of these changes?**

This PR re-organizes the clustering module README file per #294, to mirror that of the GOI module [README](https://github.com/AlexsLemonade/scpca-downstream-analyses/tree/main/optional-goi-analysis#readme).

This PR also adds a section under expected output mentioning the clustering results added to the colData of the object and how to access these results (per #240).

**What changes did you make?**

The `optional-clustering-analysis/README.md` file is re-organized here to follow the structure seen in the GOI docs which includes moving the table of contents up, then following that with the analysis overview, expected input, parameters and config file, and expected output sections.

"What to expect" in the output was also added in under the expected output section to outline what was added to the colData of the SingleCellExperiment objects (noting here that we did not add anything to the metadata of the SCEs in the clustering workflow).

**Any comments, concerns, or questions important for reviewers**
- Do these changes make sense to you? Do they seem to address both #294 and #240?

(For reference, this is stacked on PR #293 since some clustering docs changes were made there as well)

**Checklist**

Place an `x` in all boxes that you have completed.

- [ ] I have run the most recent version of the code
- [ ] If I am adding in reports or generating any plots, I have reviewed the necessary reports
- [x] I have added all necessary documentation (if applicable)

### CODE REVIEW

File: optional-clustering-analysis/README.md [hunk 1]
```diff
@@ -29,6 +19,17 @@ For a more in depth discussion on these metrics and how they can be used to iden
 
 ## Analysis overview
 
+Libraries are unique, which means that the optimal clustering is likely to be library-dependent.
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:22]:
```suggestion
Single-cell gene expression libraries are unique, which means that the optimal clustering is likely to be library-dependent.
```

File: optional-clustering-analysis/README.md [hunk 2]
```diff
@@ -42,6 +43,24 @@ R 4.2 is required for running our pipeline, along with Bioconductor 3.15.
 Package dependencies for the analysis workflows in this repository are managed using [`renv`](https://rstudio.github.io/renv/index.html), which must be installed locally prior to running the workflow.
 If you are using conda, dependencies can be installed as [part of the initial setup](../README.md#snakemakeconda-installation).
 
+## Running the workflow
+
+The execution file with the clustering Snakemake workflow is named cluster.snakefile and can be found in the root directory. To tell snakemake to run the specific clustering workflow be sure to use the --snakefile or -s option followed by the name of the snakefile, cluster.snakefile. After navigating to within the root directory of the scpca-downstream-analyses repository, the below example command can be used to run the clustering workflow:
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:48]:
```suggestion
The execution file with the clustering Snakemake workflow is named `cluster.snakefile` and can be found in the root directory. To tell snakemake to run the specific clustering workflow be sure to use the `--snakefile` or `-s` option followed by the name of the snakefile, `cluster.snakefile`. 
After navigating to within the root directory of the `scpca-downstream-analyses` repository, the below example command can be used to run the clustering workflow:
```

File: optional-clustering-analysis/README.md [hunk 3]
```diff
@@ -118,4 +119,27 @@ The parameters found in the `config/cluster_config.yaml` file can be optionally
...
 |[View Clustering Config File](../config/cluster_config.yaml)|
-|---|
\ No newline at end of file
+|---|
+
+## Expected output
+
+For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return five files in the same directory where the inputted RDS file is stored:
+
+1. The `SingleCellExperiment` RDS file containing the added clustering results that were calculated in the first step of the worflow.
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:128]: We should include the expected name of this file.

File: optional-clustering-analysis/README.md [hunk 4]
```diff
@@ -118,4 +119,27 @@ The parameters found in the `config/cluster_config.yaml` file can be optionally
...
+2. The `_clustering_report.html` file, which is the summary html report with plots containing the clustering results.
+3. A `_clustering_all_validity_stats.tsv` file with all of the calculated cluster validity statistics.
+4. A `_clustering_summary_validity_stats.tsv` file with the cluster validity stats averaged across each cluster assignment.
+5. A `_clustering_summary_stability_stats.tsv` file with the average adjusted Rand Index (ARI) values across each cluster assignment.
+
+The TSV files will be saved in a subdirectory called `{library_id}_{filtering_method}_clustering_stats/`, where both the `{library_id}` and `{filtering_method}` are obtained from the input metadata file.
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:134]: Instead of this line, should we include a tree structure of the files?

**Review by allyhawkins — COMMENTED** [review]
This looks generally good, just a few typos and some suggestions on updating the output section.

File: optional-clustering-analysis/README.md (hunk 4 above)
Comment (cbethell) [reply optional-clustering-analysis/README.md]: Oh yes, good idea!

**Comment by cbethell** [comment]
Thanks for the review @allyhawkins! I've implemented those changes so let me know what you think of things now!

File: optional-clustering-analysis/README.md [hunk 5]
```diff
@@ -118,4 +120,38 @@ The parameters found in the `config/cluster_config.yaml` file can be optionally
 | `overwrite_results` | a binary value indicating whether or not to overwrite any existing clustering results | `TRUE` |
 
 |[View Clustering Config File](../config/cluster_config.yaml)|
-|---|
\ No newline at end of file
+|---|
+
+## Expected output
+
+For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return five files in the same directory where the inputted RDS file is stored:
```
Comment (allyhawkins) [inline optional-clustering-analysis/README.md:127]:
```suggestion
For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return five files in the same directory where the input RDS file is stored:
```

**Review by allyhawkins — APPROVED** [review]
This looks good! Just one minor edit 👍

**Pull request merged**